app/src/main/java/camp/nextstep/edu/github/MainActivity.kt

```diff
@@ -1,11 +1,47 @@
 package camp.nextstep.edu.github
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.View.GONE
+import android.view.View.VISIBLE
+import android.widget.Toast
+import android.widget.Toast.LENGTH_SHORT
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Observer
+import camp.nextstep.edu.github.databinding.ActivityMainBinding
+import camp.nextstep.edu.github.mapper.toExceptionMessage
+import camp.nextstep.edu.github.model.ExceptionMessageModel
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+
+@AndroidEntryPoint
+internal class MainActivity : AppCompatActivity() {
+
+    @Inject lateinit var repositoryAdapter: RepositoryAdapter
+    private val viewModel: MainViewModel by viewModels()
+
+    private lateinit var binding: ActivityMainBinding
 
-class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
+        binding.lifecycleOwner = this
+        binding.repositoryAdapter = repositoryAdapter
+
+        viewModel.repositorySate.observe(this, Observer {
+            binding.progressCircular.visibility = GONE
```

Comment:
코틀린 람다식을 활용해서 아래처럼 작성해보는 것은 어떨까요?
파라미터 마지막의 람다식은 중괄호로 빼는 것이 코틀린 컨벤션입니다.
```suggestion

        viewModel.repositorySate.observe(this) {
            binding.progressCircular.visibility = GONE
            ....
```